pkg/apis/batch/validation/validation.go

```diff
@@ -546,6 +548,31 @@ func ValidateJobTemplateSpec(spec *batch.JobTemplateSpec, fldPath *field.Path, o
 	return allErrs
 }
 
+func validateCompletions(spec, oldSpec batch.JobSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// Completions is immutable for non-indexed jobs.
+	// For Indexed Jobs, if ScalableIndexedJob feature gate is not enabled,
+	// fall back to validating that spec.Completions is always immutable.
+	isIndexedJob := spec.CompletionMode != nil && *spec.CompletionMode == batch.IndexedCompletion
+	if !isIndexedJob || !feature.DefaultFeatureGate.Enabled(features.ScalableIndexedJob) {
+		return apivalidation.ValidateImmutableField(spec.Completions, oldSpec.Completions, fldPath)
+	}
+
+	// Indexed Jobs cannot set completions to nil. The nil check
+	// is already performed in validateJobSpec, no need to add another error.
+	if spec.Completions == nil || oldSpec.Completions == nil {
+		return allErrs
+	}
+
+	// For Indexed Jobs, spec.Completions can only be mutated in tandem
+	// with spec.Parallelism such that spec.Completions == spec.Parallelism
+	if *spec.Completions != *oldSpec.Completions && *spec.Completions != *spec.Parallelism {
+		allErrs = append(allErrs, field.Invalid(fldPath, spec.Completions, "spec.Completions can only be modified in tandem with spec.Parallelism"))
```

Comment:
We need to check that completions = parallelism for both the old and new to allow to mutate completions.

```
if *spec.Completions != *oldSpec.Completions {
  if *spec.Completions != *spec.Parallelism || *oldSpec.Completions != *oldSpec.Parallelism {
      allErrs = append(allErrs, field.Invalid(fldPath, spec.Completions, fmt.Sprintf("can only be modified in tandem with %s", fldPath.Root("parallelism").String()))
  }
}
```

Reply:
Done